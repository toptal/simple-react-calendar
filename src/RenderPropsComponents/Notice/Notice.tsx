import React, { FC } from 'react'

type Props = {
  blockClassName?: string
}

const displayName = 'Notice'

const Notice: FC<Props> = ({ blockClassName, children }) => {
  if (!children) return null

  return <div className={`${blockClassName}-notice`}>{children}</div>
}

Notice.displayName = displayName

export default Notice
