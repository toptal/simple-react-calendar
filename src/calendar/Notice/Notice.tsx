import React, { FC } from 'react'

import { INoticeType } from '../../@types'

export interface Props {
  blockClassName: string
  type: INoticeType
}

const Notice: FC<Props> = ({type, blockClassName}) => {
  const content =
    type === 'overlapping_with_disabled' ||
    (type === 'disabled_day_click' && 'Selected range must not overlap with disabled dates.')

  return <div className={`${blockClassName}-notice`}>{content}</div>
}

Notice.displayName = 'Notice'

export default Notice
