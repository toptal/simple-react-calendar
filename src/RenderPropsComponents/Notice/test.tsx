import { render } from '@testing-library/react'
import React, { ComponentProps } from 'react'

import Notice from './Notice'

const renderNotice = (props: Partial<ComponentProps<typeof Notice>> = {}) =>
  render(<Notice {...props} />)

test('default render', () => {
  const { container } = renderNotice({ children: 'Example notice error', blockClassName: 'example-class-name' })

  expect(container).toMatchSnapshot()
})

describe('when no `children` defined', () => {
  test('renders null', () => {
    const { container } = renderNotice({ })

    expect(container).toMatchSnapshot()
  })
})
