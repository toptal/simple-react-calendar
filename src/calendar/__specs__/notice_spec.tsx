import { shallow } from 'enzyme'
import React from 'react'

import Notice from '../notice'

describe('Notice', () => {
  let wrapper, props

  beforeEach(() => {
    props = getProps()
    wrapper = shallow(<Notice {...props} />)
  })

  describe('when prop `type` is "overlapping_with_disabled"', () => {
    it('renders <Notice />', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when prop `type` is "disabled_day_click"', () => {
    it('renders <Notice />', () => {
      wrapper.setProps('disabled_day_click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})

const getProps = (overrides = {}) => ({
  type: 'overlapping_with_disabled',
  blockClassName: 'example-class-name',
  ...overrides,
})
