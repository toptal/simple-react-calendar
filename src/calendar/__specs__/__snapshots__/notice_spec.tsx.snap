// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice when prop \`type\` is "disabled_day_click" renders <Notice /> 1`] = `
<div
  className="example-class-name-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`;

exports[`Notice when prop \`type\` is "overlapping_with_disabled" renders <Notice /> 1`] = `
<div
  className="example-class-name-notice"
>
  Selected range must not overlap with disabled dates.
</div>
`;
